cmd/tendermint/commands/rollback_test.go

```diff
@@ -0,0 +1,63 @@
+package commands_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/tendermint/cmd/tendermint/commands"
+	"github.com/tendermint/tendermint/rpc/client/local"
+	rpctest "github.com/tendermint/tendermint/rpc/test"
+	e2e "github.com/tendermint/tendermint/test/e2e/app"
+)
+
+func TestRollbackIntegration(t *testing.T) {
+	dir := t.TempDir()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	cfg, err := rpctest.CreateConfig(t.Name())
+	require.NoError(t, err)
+	cfg.BaseConfig.DBBackend = "goleveldb"
+
+	subctx, subcancel := context.WithCancel(ctx)
+	defer subcancel()
+	app, err := e2e.NewApplication(e2e.DefaultConfig(dir))
+	require.NoError(t, err)
+	node, _, err := rpctest.StartTendermint(subctx, cfg, app, rpctest.SuppressStdout)
+	require.NoError(t, err)
```

Comment:
Consider maybe putting these sections into subtests, so that the context only needs one scope?

```go
t.Run("Setup", func(t *testing.T) {
  sctx, cancel := context.WithCancel(ctx)
  defer cancel()
  // blah, blah
  node.Wait()
})
```

and similarly below. Not that we strictly need a subtest, but it would make the namespace a little easier to follow without `x`, `x2`, etc. naming.